enhancements/sig-architecture/11-cluster-deletion/README.md

```diff
@@ -0,0 +1,122 @@
+
+## Summary
+
+Currently, when we remove a cluster from the Hub, the `managedClusterAddons` and `manifestWorks` in the cluster namespace need to be deleted manually, and cannot ensure that the resources deployed on the Spoke cluster can be cleaned up. This proposal is to define clearly what happened when a cluster is deleted.
+
+## Motivation
+
+Today, when we delete a `managedCluster`, only the `rolebinding` for the registration/work agent is cleaned up on the hub, and the resources including the cluster namespace,`managedClusterAddons` and `manifestWorks` are left over. It will bring some issues like: 
+1. We cannot delete `manifestWork` directly because the finalizer cannot be removed by the work-agent.
+2. The resources deployed by `manifestWork` on the spoke cluster cannot be cleaned up even if its `manifestWork` is deleted.
+3. The `managedClusterAddon` with pre-delete job cannot be deleted because the `manifestWorks` cannot be deleted too.
+
+There is a typical use case that the lifecycle of the `klusterlet` on the spoke cluster is taken over by the `manifestWork` too. There is a `manifestWork` to deploy klusterlet manifests itself after the cluster is registered. And the klusterlet manifests of the managed cluster will be cleaned up if the klusterlet manifestWork is deleted on the hub.
+The cleanup starts when the `managedCluster` is deleting.
+
+1. Delete the `managedClusterAddons` and `manifestWorks` firstly except the `klusterlet manifestWork`.
+2. Delete the `klusterlet manifestWork` after the other `managedClusterAddons` and `manifestWorks` are deleted.
+3. Start to delete the cluster namespace when there is no `managedClusterAddons` and `manifestWorks`. The cluster ns will be deleted after the resources related with Hive like job, clusterDeployment, etc are deleted. Because the Hive will create some jobs in this ns when the provisioned cluster is destroyed.
+
+The deletion process is a typical use case and example for the community. This proposal is to enhance the cluster deletion process in OCM.
+
+### Goals
+
+- Enhance the cluster deletion process and clean up the resources including `managedClusterAddons`, `manifestWorks` and namespace in order.
+
+### Non-Goals
+
+- Clean up customized resources automatically.
+
+## Proposal
+
+### Design Details 
+#### registration-controller 
+
+1. Add a condition type `ContentDeleteSuccess` to the `managedCluster`, which indicates whether the cluster resources are remaining during the `managedCluster` deletion process.
+
+2. Define a label named `cluster.open-cluster-management.io/delete-by-other`. The cluster resources with this label will not be deleted by the ACM during the `managedCluster` deletion process.
+   
+3. Add options for the registration-controller binary. The registration-controller will handle the deletion process.
+   ```
+      /registrion
+      -- controller
+      -- forceDeletionGracePeriodSeconds: 600
```

Comment:
why do we need this force delete?

Reply:
if the spoke cluster is destroyed before start to delete the managedCluster, the manifestWorks cannot be deleted because there is on one to remove the finalizer on them.
in this case, we need force delete the manifestWorks when the cluster is unavailable for a grace period, like 10min.

Reply:
I do not think we can diff between disconnected or cluster is down. Registration may not fit to handle this case, it may need higher level scheduler to move the workload. I am thinking maybe we should not let registration to delete work, but only addon (addon is managed by registration). registration should only check if work is cleaned already.

Reply:
updated as we discussed.